llvm/include/llvm/Support/KnownBits.h

```diff
@@ -354,6 +354,18 @@ struct KnownBits {
   /// Compute knownbits resulting from llvm.usub.sat(LHS, RHS)
   static KnownBits usub_sat(const KnownBits &LHS, const KnownBits &RHS);
 
+  /// Compute knownbits resulting from (C1 & C2) + (C1 ^ C2).ashr(1)
```

Comment:
Putting the bit twidling formula in the comment is not helpful. Suggestion:
```suggestion
  /// Compute knownbits resulting from APIntOps::avgFloorS
```
Same for the others.